### ISSUE TITLE
When intercepting a route, boat lurks when reaching guide radius

### ISSUE THREAD

**Issue by marcobergman** [seandepagnier/autopilot_route_pi#37]
When intercepting a route, the boat maintains max-error-angle until the moment it is the guide radius away from the route. At that moment it suddenly lurks to a heading perpendicular to the route, no longer maintaining max-error-angle. 

![image](https://github.com/seandepagnier/autopilot_route_pi/assets/17980560/fdc9c3d5-6aa2-4a1f-983d-f363a6214961)

This seems to be caused by the change in the latest commit of jan 18th:

![image](https://github.com/seandepagnier/autopilot_route_pi/assets/17980560/1520aad2-6863-4e78-9271-0e6cc392d322)

Please remove this; I don't want my boat to make this type of movements. This was observed in the 0.4.30 version from the master plugin catalog on OpenCPN 5.8.4, Ubuntu 20.04, in a simulated environment.

**Comment by marcobergman** [comment]
It still occurs in 0.4.31, and by now I have also encountered this in real life under sail, with very unpleasant effects.

![image](https://github.com/seandepagnier/autopilot_route_pi/assets/17980560/a540fb0a-435c-467c-9c48-70ee202486df)

**Comment by marcobergman** [comment]
![image](https://github.com/seandepagnier/autopilot_route_pi/assets/17980560/c93e9d24-f17e-44fe-ad32-96e8f187a49b)

**Comment by seandepagnier** [comment]
I have this issue in mind,  sorry I am focused on other projects but I will certainly look into it and do some sailing tests soon myself.

**Comment by seandepagnier** [comment]
I had made this change a while ago but it did not get committed..    hopefully it is fixed now

**Comment by marcobergman** [comment]
Looks good, I'll pull it through the simulation later. What's the extra
functionality added?

**Comment by marcobergman** [comment]
Indeed, the original 'lurking' problem seems to be solved. The functionality of the newly introduced 'Intersect' checkbox is not clear, though. When it is 'off', the route is intercepted with MaxErrorAngle (orange track). When it is 'on', the interception angle seems to be higher (yellow track), possibly double. I've seen the code but could not explain it in words. In any case, I'd suggest the intersection checkbox to be 'off' by default to preserve the meaning of MaxErrorAngle.

![image](https://github.com/seandepagnier/autopilot_route_pi/assets/17980560/850aa4a5-d089-4009-8c28-6d954a0b3d49)

Btw the spinners on MaxErrorAngle and Distance are now (partly) hidden (compiled on Ubuntu 22.04), see above picture.
